### PR TITLE
Scan and charge

### PR DESCRIPTION
**Pull request by Skyrix06**
**_site to encode data in base64 to test:_** 

https://www.base64encode.net/

**_site to generate QR Code:_** 

https://fr.qr-code-generator.com/solutions/app-qr-code/?ut_source=google_c&ut_medium=cpc&ut_campaign=franzoesisch_dsa&ut_content=solutions&ut_term=_b&gclid=CjwKCAjwiaX8BRBZEiwAQQxGx7RgAQsmBBB2ZZQHmxInYYLWLstRSjLA71QpZ-SPrVnj4hFALg76xBoCh3UQAvD_BwE

Data syntax exemple:

{"tenantSubDomain":"slf", "tenantName":"SAP Labs France (prod)", "endpoint":"scp", "chargingStationID":"SAP-Mougins-18", "connectorID":2}

### CODE REVIEW

File: ios/PodFile [hunk 1]
```diff
@@ -23,6 +23,10 @@ target 'eMobility' do
   pod 'GoogleMaps'
   pod 'Google-Maps-iOS-Utils'
 
+  # Camera Permission
+  permissions_path = '../node_modules/react-native-permissions/ios'
+  pod 'Permission-Camera', :path => "#{permissions_path}/Camera.podspec"
```
Comment (jerome-benoit) [inline ios/PodFile:28]: Are you sure you need to add it? 
It should be automatically handled by the related dependency with RN 0.6x

File: package.json [hunk 2]
```diff
@@ -90,7 +90,9 @@
     "react-native-actionsheet": "^2.4.2",
     "react-native-animatable": "^1.3.3",
     "react-native-appearance": "^0.3.4",
+    "react-native-base64": "^0.1.0",
     "react-native-calendars": "^1.403.0",
+    "react-native-camera": "^3.40.0",
```
Comment (jerome-benoit) [inline package.json:95]: No related @types dep?

File: package.json [hunk 3]
```diff
@@ -109,6 +111,8 @@
     "react-native-modalize": "^2.0.6",
     "react-native-orientation-locker": "^1.2.0",
     "react-native-paper": "^3.11.0",
+    "react-native-permissions": "^2.2.2",
+    "react-native-qrcode-scanner": "^1.4.1",
```
Comment (jerome-benoit) [inline package.json:115]: No related @types dep?

File: src/I18n/languages/de.json [hunk 4]
```diff
@@ -128,6 +129,19 @@
     "reportErrorSuccess": "Der Fehler wurde erfolgreich gemeldet",
     "reportErrorFailed": "Der Fehler kann nicht gemeldet werden"
   },
+  "qrCode": {
+    "wrongOrganizationTitle": "Change Organization",
+    "wrongOrganizationMessage": "The charging Station you try to access is not on your organization do you want to switch to {{tenantName}}?",
```
Comment (jerome-benoit) [inline src/I18n/languages/de.json:134]: Missing comma at 'organization'

File: src/I18n/languages/en.json [hunk 5]
```diff
@@ -50,7 +50,8 @@
     "lastChangedBy": "Last Changed By",
     "mobile": "Mobile",
     "errorTitle": "Error title",
-    "errorDescription": "Description"
+    "errorDescription": "Description",
+    "scanQrCodeSuccess": "The QR code has been scan successfully"
```
Comment (jerome-benoit) [inline src/I18n/languages/en.json:54]: scan -> scanned

File: src/I18n/languages/es.json [hunk 6]
```diff
@@ -128,6 +129,19 @@
     "reportErrorSuccess": "El error se ha informado correctamente",
     "reportErrorFailed": "No se puede informar el error"
   },
+  "qrCode": {
+    "wrongOrganizationTitle": "Change Organization",
+    "wrongOrganizationMessage": "The charging Station you try to access is not on your organization do you want to switch to {{tenantName}}?",
```
Comment (jerome-benoit) [inline src/I18n/languages/es.json:134]: Idem

**Review by jerome-benoit — CHANGES_REQUESTED** [review]
What is the data structure expected to be encoded in the QR code? 
A simple URI? a formatted string?

File: ios/PodFile (hunk 1 above)
Comment (Skyrix06) [reply ios/PodFile:28]: Data syntax exemple:

{"tenantSubDomain":"slf", "tenantName":"SAP Labs France (prod)", "endpoint":"scp", "chargingStationID":"SAP-Mougins-18", "connectorID":2}

Comment (Skyrix06) [reply ios/PodFile:28]: Try to remove it and see if anything wrong I will push with the modification if it work without

File: package.json (hunk 3 above)
Comment (Skyrix06) [reply package.json:115]: not as I see on the documentation:

https://github.com/moaazsidat/react-native-qrcode-scanner

File: package.json (hunk 2 above)
Comment (Skyrix06) [reply package.json:95]: Not has I see on the documentation:

https://github.com/react-native-camera/react-native-camera

**Comment by jerome-benoit** [comment]
> Data syntax exemple:
> 
> {"tenantSubDomain":"slf", "tenantName":"SAP Labs France (prod)", "endpoint":"scp", "chargingStationID":"SAP-Mougins-18", "connectorID":2}

Could you please add a qrcode-template.json file somewhere in assets directory?

File: ios/PodFile (hunk 1 above)
Comment (Skyrix06) [reply ios/PodFile:28]: It's not an autolink, but a dependency for the permission camera on IOS, I've done some test on my side and I see that without this line of code the permission camera popup doesn't display on IOS

Comment (jerome-benoit) [reply ios/PodFile:28]: Ok.

**Pull request merged**